cmd/api/src/database/agi.go

```diff
@@ -86,47 +86,74 @@ func (s *BloodhoundDB) DeleteAssetGroup(ctx context.Context, assetGroup model.As
 	})
 }
 
+func (s *BloodhoundDB) getAssetGroupMemberCount(ctx context.Context, assetGroupID int32) (int, error) {
+    var (
+        latestCollection model.AssetGroupCollection
+        result           = s.preload(model.AssetGroupCollectionAssociations()).
+                             WithContext(ctx).
+                             Where("asset_group_id = ?", assetGroupID).
+                             Order("created_at DESC").
+                             First(&latestCollection)
+    )
+
+    if result.Error != nil {
+        if errors.Is(result.Error, gorm.ErrRecordNotFound) {
```

Comment:
The BloodHound `errors` package predates much of the added functionality in the [upstream errors package](https://pkg.go.dev/errors). We should prefer the stdlib `errors.Is(...)` instead of our implementation.

I'd ask for three changes here:
* Refactor this without a `nil` check since `errors.Is(...)` is nil-safe.
* Consume the [stdlib errors package](https://pkg.go.dev/errors) in the conditional.
* Add a deprecation notice to the BloodHound `errors` function to help guard others from exercising it.